src/util.js

```diff
@@ -107,3 +107,80 @@ export function format_memory_and_limit(usage, limit) {
         return _("");
     }
 }
+
+export function updateContainers() {
+    let newContainers = {};
+    let newContainersStats = {};
+    let newContainersMeta = [];
+    let inspectRet = [];
+    let containerStatsRet = [];
+    let countRun;
+    return new Promise((resolve, reject) => {
+        varlinkCall(PODMAN, "io.podman.ListContainers")
+                .then(reply => {
+                    newContainersMeta = reply.containers;
+                    if (!newContainersMeta) {
+                        console.log(newContainersMeta);
+                        resolve({newContainers: newContainers, newContainersStats: newContainersStats});
+                    }
+                    inspectRet = newContainersMeta.map((container) => {
+                        return new Promise((resolve, reject) => {
+                            varlinkCall(PODMAN, "io.podman.InspectContainer", {name: container.id})
+                                    .then(reply => {
+                                        resolve(JSON.parse(reply.container));
+                                    })
+                                    .catch(ex => reject(new Error("Failed to do InspectContainer call:", ex, JSON.stringify(ex))));
+                        });
+                    });
+                })
+                .then(function() {
+                    const containerCount = newContainersMeta.length;
+                    Promise.all(inspectRet)
+                            .then((inspectRet) => {
+                                let runEle = newContainersMeta.filter((ele) => {
+                                    return ele.status === "running";
+                                });
+                                inspectRet.map((inspectRet) => {
+                                    newContainers[inspectRet.ID] = inspectRet;
+                                });
+                                if (runEle.length === 0 && Object.keys(newContainers).length === containerCount) {
+                                    resolve({newContainers: newContainers, newContainersStats: newContainersStats});
+                                }
```

Comment:
This condition is not easy to understand. As for the length comparison, how would that ever be false? `Promise.all.then()` already guarantees that all promises succeeded and thus `newContainers` has the same number of elements as `containerCount`. I think you can just drop that condition and the `containerCount` variable. 

In general, if you have to do this kind of explicit length comparison with data type iteration, then this smells like there's something wrong with your data structures or your iteration. These seem alright here, and the condition is just superflous.

Likewise, if there are no running containers, you resolve the returned promise here, but *don't* return the handler. The code below will still run, no? (See the above "hard to understand the structure and then ladder"). 

I think the code should still be correct and much easier to understand if the `Promise.all.then` handler consists only of the above `inspectRet.map()` call.